device/transport/mqtt/src/mqtt.ts

```diff
@@ -1020,6 +1020,16 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
       });
     }
   }
+
+  private _ignoreConnectionClosedInErrorCallback(callback: (err?: Error, ...args: any[]) => void): (err?: Error, ...args: any[]) => void {
+    return (err: Error, ...args: any[]) => {
+      if (err?.name === 'Error' && err?.message === 'Connection closed') {
+        debug('Mqtt subscribe/unsubscribe operation failed due to MQTT.js connection closed error. MqttBase will handle this when MQTT.js emits the close event.');
+        return;
+      }
+      callback(err, ...args);
+    };
+  }
```

Comment:
We want to avoid connection closed errors because we don't want these errors to bubble up to the user through the `error` event path.

MQTT.js will emit a close event, and our transport code will handle it accordingly.